.github/workflows/release-staging.yaml

```diff
@@ -0,0 +1,193 @@
+---
+name: 'Release: Staging'
+
+on:
+  push:
+    branches:
+      - future
+
+jobs:
+  verify:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cloning repository
+        uses: actions/checkout@v2
+
+      - name: Setting up Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: node_modules-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Installing Node.js packages
+        run: npm install
+
+      - name: Running tests
+        run: |
+          gulp updateTestResources
+          gulp buildComponentVersions
+          gulp buildPixiFunctions
+          gulp lintNode
+          gulp lintYaml
+          npm run test:platform
+          npm run test:playground
+          npm run test:pixi
+  prepare:
+    needs: verify
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cloning repository
+        uses: actions/checkout@v2
+
+      - name: Setting up Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: node_modules-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Installing Node.js packages
+        run: npm install
```

Comment:
Jep, but jobs (verify, prepare, build, release in this case) don't share VMs but only steps (Cloning, Running tests, ...) do. Hence cloning the repository and installing the dependencies is done multiple times.

It isn't ideal regarding run-times but seems to be best practice, i.e. using distinct jobs instead of having one giant job with more steps.

Reply:
Got it. Makes sense